addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,263 @@
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("Bridge Status updated to {} for device: {}", bridgeStatusInfo.getStatus(), getThing().getUID());
+        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, bridgeStatusInfo.getDescription());
+        }
+
+    }
+
+    /**
+     * Requests the corresponding meter data and stores the meterId internally for later usage
+     *
+     * @param token The current active auth token
+     */
+    private void obtainMeterId(String token) {
+        try {
+
+            String url = getApiString(API_METER_ENDPOINT);
+
+            Properties urlHeader = new Properties();
+            urlHeader.put("CONTENT-TYPE", "application/json");
+            urlHeader.put("Authorization", token);
+
+            String urlResponse = HttpUtil.executeUrl("GET", url, urlHeader, null, null, 2000);
+            JsonObject responseJson = (JsonObject) new JsonParser().parse(urlResponse);
+
+            if (responseJson.has("meter_id")) {
+                setMeterID(responseJson.get("meter_id").toString());
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+
+            String errorMsg = new StringBuilder("Invalid Api Response ( ").append(responseJson.toString()).append(" )")
+                    .toString();
+
+            throw new Exception(errorMsg);
+
+        } catch (Exception e) {
+            String errorMsg = new StringBuilder("Could not initialize Thing")
+                    .append(System.getProperty("line.separator")).append(this.getThing().getUID())
+                    .append(System.getProperty("line.separator")).append(e.getMessage()).toString();
+
+            logger.debug(errorMsg, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * requests a pre-filtered reading list for this specific meter and updates all corresponding channels
+     *
+     * @param token The current active access token
+     */
+    private void updateMeter(String token) {
+        try {
+            String url = getApiString(API_READINGS_ENDPOINT);
+
+            Properties urlHeader = new Properties();
+            urlHeader.put("CONTENT-TYPE", "application/json");
+            urlHeader.put("Authorization", token);
+
+            String urlResponse = HttpUtil.executeUrl("GET", url, urlHeader, null, null, 2000);
+
+            ReadingInstance latestReading = gson.fromJson(new JsonParser().parse(urlResponse), ReadingInstance.class);
+
+            // UoM - Update Quantity State depending on the thing type.
+
+            ThingTypeUID thingtype = getThing().getThingTypeUID();
+
+            if (thingtype.equals(THING_TYPE_ENERGYMETER)) {
```

Comment:
Prefer to use the constant on the left side of equals. Because if the variable is null this code crashes, while with the constant on the left side it will run just fine.